access/ms-teams/ms_teams_test.go

```diff
@@ -0,0 +1,677 @@
+	}
+
+	reviewer := s.mockAPI.StoreUser(msapi.User{Mail: s.userNames.reviewer1})
+
+	s.startApp()
+
+	req := s.createAccessRequest([]msapi.User{reviewer})
+	s.checkPluginData(req.GetName(), func(data interface{}) bool {
+		return len(data.(PluginData).TeamsData) > 0
+	})
+
+	msg, err := s.mockAPI.CheckNewMessage(s.Context())
+	require.NoError(t, err)
+	require.Equal(t, reviewer.ID, msg.RecipientID)
+
+	err = s.reviewer1().SubmitAccessRequestReview(s.Context(), req.GetName(), types.AccessReview{
+		Author:        s.userNames.reviewer1,
+		ProposedState: types.RequestState_APPROVED,
+		Created:       time.Now(),
+		Reason:        "okay",
+	})
+	require.NoError(t, err)
+
+	reply, err := s.mockAPI.CheckMessageUpdate(s.Context())
+	require.NoError(t, err)
+
+	require.Equal(t, msg.RecipientID, reply.RecipientID)
+	require.Equal(t, msg.ID, reply.ID)
+	require.Equal(t, gjson.Get(reply.Body, "attachments.0.content.body.4.facts.0.value").String(), "✅")
+	require.Equal(t, gjson.Get(reply.Body, "attachments.0.content.body.4.facts.1.value").String(), s.userNames.reviewer1)
+	require.Equal(t, gjson.Get(reply.Body, "attachments.0.content.body.4.facts.3.value").String(), "okay")
+
+	err = s.reviewer2().SubmitAccessRequestReview(s.Context(), req.GetName(), types.AccessReview{
+		Author:        s.userNames.reviewer2,
+		ProposedState: types.RequestState_DENIED,
+		Created:       time.Now(),
+		Reason:        "not okay",
+	})
+	require.NoError(t, err)
+
+	reply, err = s.mockAPI.CheckMessageUpdate(s.Context())
+	require.NoError(t, err)
+
+	require.Equal(t, msg.RecipientID, reply.RecipientID)
+	require.Equal(t, msg.ID, reply.ID)
+	require.Equal(t, gjson.Get(reply.Body, "attachments.0.content.body.5.facts.0.value").String(), "❌")
+	require.Equal(t, gjson.Get(reply.Body, "attachments.0.content.body.5.facts.1.value").String(), s.userNames.reviewer2)
+	require.Equal(t, gjson.Get(reply.Body, "attachments.0.content.body.5.facts.3.value").String(), "not okay")
+}
+
+func (s *TeamsSuite) TestApprovalByReview() {
+	t := s.T()
+
+	if !s.teleportFeatures.AdvancedAccessWorkflows {
+		t.Skip("Doesn't work in OSS version")
+	}
+
+	reviewer := s.mockAPI.StoreUser(msapi.User{Mail: s.userNames.reviewer1})
+
+	s.startApp()
+
+	req := s.createAccessRequest([]msapi.User{reviewer})
+	msg, err := s.mockAPI.CheckNewMessage(s.Context())
+	require.NoError(t, err)
+	require.Equal(t, reviewer.ID, msg.RecipientID)
+
+	err = s.reviewer1().SubmitAccessRequestReview(s.Context(), req.GetName(), types.AccessReview{
+		Author:        s.userNames.reviewer1,
+		ProposedState: types.RequestState_APPROVED,
+		Created:       time.Now(),
+		Reason:        "okay",
+	})
+	require.NoError(t, err)
+
+	reply, err := s.mockAPI.CheckMessageUpdate(s.Context())
+	require.NoError(t, err)
+
+	require.Equal(t, msg.RecipientID, reply.RecipientID)
+	require.Equal(t, msg.ID, reply.ID)
+	require.Equal(t, gjson.Get(reply.Body, "attachments.0.content.body.4.facts.0.value").String(), "✅")
```

Comment:
This paths look a bit cryptic
I think we should move them to consts with a clarifying name
What do you think?

Reply:
XSLT of the new era :-) I agree, that would increase readability.